storage/blockdevice/COMPONENT_QSPIF/source/QSPIFBlockDevice.cpp

```diff
@@ -1469,6 +1481,14 @@ int QSPIFBlockDevice::_qspi_send_read_sfdp_command(mbed::bd_addr_t addr, mbed::s
         return status;
     }
 
+    // Handle CR3NV[1] quirk.
+    const mbed::bd_addr_t register_CR3NV = 0x000004;
+    if (_CR3NV_quirk && (addr == register_CR3NV)) {
+        // If we reach here, rx_buffer is guaranteed to be non-null
+        // because it's been checked by _qspi.read() above.
+        static_cast<uint8_t*>(rx_buffer)[0] |= 0x01;
```

Comment:
We should add a unit test to catch this when our mocked QSPI device has a certain device ID and when it doesn't.

Reply:
We currently don't have unit tests for QSPIFBlockDevice. I'll create a test cpp file and add a test for this quirk (i.e. what we change in this PR).

Reply:
Unit test added for the quirk. I eventually did it differently, because what we discussed didn't quite work - see commit message and inline comments for details.